xl/player/gst/dynamic_sink.py

```diff
@@ -177,5 +179,19 @@ def _add_audiosink(self, audio_sink, buffer_position):
 
         self.audio_sink = audio_sink
 
+    def set_volume(self, volume: float) -> None:
+        if not hasattr(self.audio_sink.props, 'volume'):
+            self._volume = volume
```

Comment:
This should always be stored, that way `get_volume` can never return a stale volume?